toast-tk-dao/src/main/java/io/toast/tk/dao/service/dao/access/plan/TestPlanDaoService.java

```diff
@@ -12,75 +12,79 @@
 import io.toast.tk.dao.domain.impl.report.Campaign;
-import io.toast.tk.dao.domain.impl.report.Project;
+import io.toast.tk.dao.domain.impl.report.TestPlanImpl;
+import io.toast.tk.dao.domain.impl.repository.ProjectImpl;
+import io.toast.tk.dao.service.dao.access.repository.ProjectDaoService;
 import io.toast.tk.dao.service.dao.common.AbstractMongoDaoService;
 import io.toast.tk.dao.service.dao.common.CommonMongoDaoService;
 import io.toast.tk.dao.service.init.DbStarter;
 
-public class ProjectDaoService extends AbstractMongoDaoService<Project> {
+public class TestPlanDaoService extends AbstractMongoDaoService<TestPlanImpl> {
 
-	public interface Factory {
-		
-		ProjectDaoService create(final @Assisted String dbName);
+	public interface Factory extends IServiceFactory<TestPlanDaoService>{
 	}
 
 	private final CampaignDaoService cDaoService;
+	private final ProjectDaoService pDaoService;
 
 	@Inject
-	public ProjectDaoService(
+	public TestPlanDaoService(
 		final DbStarter starter,
 		final CommonMongoDaoService cService,
 		final @Assisted String dbName,
 		final @Named("default_db") String default_db,
-		final CampaignDaoService.Factory cDaoServiceFactory
+		final CampaignDaoService.Factory cDaoServiceFactory,
+		final ProjectDaoService.Factory pDaoServiceFactory
 	) {
-		super(Project.class, starter.getDatabaseByName((dbName == null ? default_db : dbName)), cService);
+		super(TestPlanImpl.class, starter.getDatabaseByName((dbName == null ? default_db : dbName)), cService);
 		this.cDaoService = cDaoServiceFactory.create(dbName);
+		this.pDaoService = pDaoServiceFactory.create(dbName);
 	}
 
-	public Project getByName(final String name) {
-		final Query<Project> query = createQuery();
+	public TestPlanImpl getByName(final String name) {
+		final Query<TestPlanImpl> query = createQuery();
 		query.field("name").equal(name);
 		return query.get();
 	}
 
-	private static class ProjectComparator implements Comparator<Project> {
+	private static class ProjectComparator implements Comparator<TestPlanImpl> {
 
-		static final Comparator<Project> INSTANCE = new ProjectComparator();
+		static final Comparator<TestPlanImpl> INSTANCE = new ProjectComparator();
 		
 		private ProjectComparator() {
 			
 		}
 		
 		@Override
-		public int compare(final Project project1, final Project project2) {
+		public int compare(final TestPlanImpl project1, final TestPlanImpl project2) {
 			return project1.getIteration() - project2.getIteration();
 		}
 	}
 	
-	public List<Project> getProjectHistory(final Project project) {
-		final Query<Project> query = createQuery();
+	public List<TestPlanImpl> getProjectHistory(final TestPlanImpl project) {
+		final Query<TestPlanImpl> query = createQuery();
 		final Criteria nameCriteria = query.criteria("name").equal(project.getName());
 		final Criteria versionCriteria = query.criteria("version").equal(project.getVersion());
 		final Criteria iterationCriteria = query.criteria("iteration").lessThan(project.getIteration());

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "iteration" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS1192)